lottie/src/main/java/com/airbnb/lottie/animation/content/ContentGroup.java

```diff
@@ -175,24 +178,40 @@ Matrix getTransformationMatrix() {
     }
 
     // Apply off-screen rendering only when needed in order to improve rendering performance.
-    boolean isRenderingWithOffScreen = lottieDrawable.isApplyingOpacityToLayersEnabled() && hasTwoOrMoreDrawableContent() && layerAlpha != 255;
+    boolean isRenderingWithOffScreen =
+        (lottieDrawable.isApplyingOpacityToLayersEnabled() && hasTwoOrMoreDrawableContent() && layerAlpha != 255) ||
+        (shadowToApply != null && lottieDrawable.isApplyingShadowToLayersEnabled() && hasTwoOrMoreDrawableContent());
+    int childAlpha = isRenderingWithOffScreen ? 255 : layerAlpha;
+
+    Canvas contentCanvas = canvas;
     if (isRenderingWithOffScreen) {
       offScreenRectF.set(0, 0, 0, 0);
-      getBounds(offScreenRectF, matrix, true);
-      offScreenPaint.setAlpha(layerAlpha);
-      Utils.saveLayerCompat(canvas, offScreenRectF, offScreenPaint);
+      getBounds(offScreenRectF, parentMatrix, true);
+      offscreenOp.alpha = layerAlpha;
+      if (shadowToApply != null) {
+        shadowToApply.applyTo(offscreenOp);
+        shadowToApply = null; // Don't pass it to children - OffscreenLayer now takes care of this
```

Comment:
In all of my years writing Java, I always thought that parameters were final 😮

Reply:
I'm to embarrassed to even start listing the surprising things I learned about programming languages I've used entirely too late :D